hub-detect/src/main/groovy/com/blackducksoftware/integration/hub/detect/workflow/codelocation/CodeLocationNameService.java

```diff
@@ -23,24 +23,144 @@
  */
 package com.blackducksoftware.integration.hub.detect.workflow.codelocation;
 
+import java.io.File;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.blackducksoftware.integration.hub.bdio.model.externalid.ExternalId;
+import com.blackducksoftware.integration.hub.detect.bomtool.BomToolGroupType;
 import com.blackducksoftware.integration.hub.detect.util.DetectFileFinder;
 
-public abstract class CodeLocationNameService {
+public class CodeLocationNameService {
+    private final Logger logger = LoggerFactory.getLogger(CodeLocationNameService.class);
+
+    public static final int MAXIMUM_CODE_LOCATION_NAME_LENGTH = 250;
     private final DetectFileFinder detectFileFinder;
 
     public CodeLocationNameService(final DetectFileFinder detectFileFinder) {
         this.detectFileFinder = detectFileFinder;
     }
 
-    public DetectFileFinder getDetectFileFinder() {
-        return detectFileFinder;
+    public String createBomCodeLocationName(final String detectSourcePath, final String sourcePath, final ExternalId externalId, final BomToolGroupType bomToolType, final String prefix, final String suffix) {
```

Comment:
Is there a reason, you collapsed the services into this one class? Or do you just prefer this approach?